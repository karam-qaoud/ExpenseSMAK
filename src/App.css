/* themain search button */
.logemailandpassword {
  margin-top: 10px;
  width: 35%;
  padding: 15px 20px;
  border-radius: 5px;
  background-color: white;
  size: 100px;
  /* border-color: black; */
  /* font-weight: bold; */
  box-shadow: 20 20px 20px -20px #333;
}
/*the second part of app name*/
#finder {
  color: blue;
}
/*position & styling for the of app*/
#header {
  text-align: left;
  margin-left: 15px;
  font-size: xx-large;
}
/*css for login card*/
.loginform {
  width: 400px;
  box-shadow: 0 0 3px 0 rgba(0, 0, 0, 0.3);
  background: #fff;
  padding: 20px;
  margin: 8% auto 0;
  text-align: center;
  margin-left: 50%;
}

/*the header inside login card*/
.loginform h1 {
  color: #1c8abd;
  margin-bottom: 15px;
}

/*all input for loginand register*/
.inputbox {
  border-radius: 20px;
  padding: 12px;
  margin: 10px 0;
  width: 90%;
  border: 1px solid #999;
  outline: none;
}

/*for favorite and read list button */
.butooon {
  color: white;
  background-color: #1c8abd;
  width: 90%;
  padding: 10px;
  border-radius: 20px;
  font-size: 15px;
  margin: 10px 0;
  border: none;
  outline: none;
  cursor: pointer;
}


#hrtwo {
  border: none;
  height: 200px;
  width: 100%;
  height: 50px;
  border-bottom: 1px solid black;
  box-shadow: 0 10px 10px -10px;
  margin: -50px auto 10px;
  margin-top: 1%;
  size: 25x;
}

/*style about links*/
#links {
  float: right;
  margin-top: -45px;
}


#search {
  text-align: center;
}

/*size of book img*/
#img {
  border-radius: 50%;
}
/*the inout inside register card*/
.reginform {
  text-align: center;
  margin-top: 7%;
  margin-left: 30%;
}
/*login button*/
.loginbutton {
  width: 10%;
  height: 10%;
  padding: 15px 20px;
  border: none;
  border-radius: 15px;
  background-color: dodgerblue;
  font-weight: bold;
  color: white;
  margin-top: 10px;
}

#inpt {
  border-radius: 25px;
  border-color: cadetblue;
  border: none;
}
.App {
  text-align: center;
}

.App-logo {
  height: 40vmin;
  pointer-events: none;
}

@media (prefers-reduced-motion: no-preference) {
  .App-logo {
    animation: App-logo-spin infinite 20s linear;
  }
}

.App-header {
  background-color: #282c34;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}

.App-link {
  color: #61dafb;
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}

/*for the input and button seacrh*/
.buttonIn {
  width: 300px;
  position: relative;
  margin-left: 500px;
}
/*input search*/
#inpt {
  margin: 0px;
  padding: 23px;
  width: 450px;
  outline: none;
  height: 46px;
  border-radius: 5px;
  border-radius: 10px;
  background-color: white;
  font-size: 15px;
}
/*the button inside input search*/
#butn {
  position: absolute;
  border-radius: 10px;
  z-index: 4;
  border: none;
  height: 40px;
  width: 100px;
  cursor: pointer;
  margin-top: 3px;
  color: white;
  background-color: #1e90ff;
  transform: translateX(345px);
  font-size: 19px;
  padding: 2px;
}

#butn:hover {
  background-color: rgb(58, 17, 207);
}

.sora {
  width: 300px;
  height: 300px;
}
#bigDiv {
  display: inline-block;
  vertical-align: top;
  width: 56%;
  height: 50%;
}

.txt {
  font-size: 20px;
  transform: translateX(0px);
  color: #fff;
  /* margin-bottom: px; */
}
#zerinmidel {
  text-align: center;
  margin-top: 50px;
}

.favpage {
  background-color: #1e90ff; /* blue */
  border: none;
  color: white;
  padding: 15px 32px;
  border-radius: 15px;
  text-align: center;
  text-decoration: none;
  display: inline-block;
  font-size: 16px;
  margin-left: 2px;
  margin-right: 2px;
}
/* style about favaret page*/
.favpage_ {
  background-color: white; /* blue */
  border: none;
  color: #1e90ff;
  padding: 15px 32px;
  border-radius: 15px;
  text-align: center;
  text-decoration: none;
  display: inline-block;
  font-size: 16px;
  margin-left: 2px;
  margin-right: 2px;
}

/* style button in rede later and favouret page */
.zer {
  background-color: #1e90ff; /* blue */
  border: none;
  color: white;
  padding: 15px 32px;
  border-radius: 15px;
  text-align: center;
  text-decoration: none;
  display: inline-block;
  font-size: 16px;
  margin-left: 2px;
  margin-right: 2px;
  margin-top: -17px;
}
/* hover*/
.zer:hover {
  background-color: rgb(44, 158, 179);
}
#wh {
  color: #fff;
}
/* style about add to fav and read later button in show search book page*/
.zeren {
  background-color: #1e90ff; /* blue */
  border: none;
  color: white;
  padding: 15px 32px;
  border-radius: 15px;
  text-align: center;
  text-decoration: none;
  display: inline-block;
  font-size: 16px;
  margin-left: 2px;
  margin-right: 2px;
  margin-top: -17px;
}
.zeren:hover {
  background-color: rgb(44, 158, 179);
}
