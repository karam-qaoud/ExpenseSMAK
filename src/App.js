import React from 'react';
import './App.css';
import Registration from './registration.js';
import Login from './login.js';
import {
  BrowserRouter as Router,
  Switch,
  Route,
  Redirect,
} from 'react-router-dom';
import App2 from './App2';

//main component
class App extends React.Component {
  constructor(props) {
    super(props);
    this.state = {};
  }
  //Authuntocation
  setUserAuth = (value) => this.setState({ isAuthenticated: true });
  //switch between login page & register page & App2 page
  render() {
    return (
      <div className='app'>
        <Router>
          <Switch>
            <Route path='/auth/login'>
              <Login setUserAuth={this.setUserAuth} />
            </Route>
            <Route exact path='/auth/reg'>
              <Registration setUserAuth={this.setUserAuth} />
            </Route>

            <PrivateRoute isAuthenticated={this.state.isAuthenticated} path='/'>
              <App2 />
            </PrivateRoute>
          </Switch>
        </Router>
      </div>
    );
  }
}
//prevent user to enter main page without login or register
function PrivateRoute({ children, isAuthenticated, ...rest }) {
  return (
    <Route
      {...rest}
      render={({ location }) =>
        isAuthenticated ? (
          children
        ) : (
          <Redirect
            to={{
              pathname: '/auth/login',
              state: { from: location },
            }}
          />
        )
      }
    />
  );
}

export default App;
