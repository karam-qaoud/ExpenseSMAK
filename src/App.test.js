import React from 'react';
// import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Showone from './showone';
import App from './App';
import { shallow } from 'enzyme';
import { mount } from 'enzyme';

it('renders my app component', () => {
  shallow(<App />);
});

test('snapshot renders', () => {
  const component = renderer.create(<App />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it('fetches async data', () => {
  const promise = new Promise((resolve, reject) =>
    setTimeout(
      () =>
        resolve({
          data: [
            {
              _id: '5f11a35b317b4246bc260ea9',
              title: '100 Poems by Faiz Ahmed Faiz, 1911-1984',
              dateOfPublication: '2002-01-01',
              img:
                'http://books.google.com/books/content?id=taSGbCRIW5cC&printsec=frontcover&img=1&zoom=5&edge=curl&source=gbs_api',
              __v: 0,
            },
            {
              _id: '5f11a373317b4246bc260eab',
              title: 'Memorials of the Hilles Family',
              dateOfPublication: '1928',
              img:
                'http://books.google.com/books/content?id=rfNUAAAAMAAJ&printsec=frontcover&img=1&zoom=5&source=gbs_api',
              __v: 0,
            },
            {
              _id: '5f11a375317b4246bc260eac',
              title:
                "Investigation of Real Estate Bondholders' Reorganizations, Public Hearings Before a Subcommittee of ... 73:2-74:2",
              dateOfPublication: '1934',
              img:
                'http://books.google.com/books/content?id=cRoVAAAAIAAJ&printsec=frontcover&img=1&zoom=5&edge=curl&source=gbs_api',
              __v: 0,
            },
          ],
        }),
      100
    )
  );
  const wrapper = mount(<Showone />);

  expect(wrapper.find('.big-div-item').length).toEqual(0);

  promise.then(() => {
    expect(wrapper.find('.big-div-item').length).toEqual(3);
  });
});
