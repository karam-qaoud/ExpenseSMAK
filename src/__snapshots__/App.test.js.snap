// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot renders 1`] = `
<div
  className="app"
>
  <div
    class="loginform"
  >
    <h1>
      Log in Now
    </h1>
    <form
      onSubmit={[Function]}
    >
      <input
        class="inputbox"
        name="email"
        onChange={[Function]}
        placeholder="Email"
        required={true}
        type="email"
        value=""
      />
      <br />
      <input
        class="inputbox"
        name="password"
        onChange={[Function]}
        placeholder="Password"
        required={true}
        type="password"
        value=""
      />
      <br />
      <button
        class="butooon"
        type="submit"
      >
        Login
      </button>
      <hr />
      <p>
        Dont have an account ? 
        <a
          href="/auth/reg"
          onClick={[Function]}
        >
           Sgin up
        </a>
         
      </p>
    </form>
  </div>
</div>
`;
